packages/cloud_firestore/lib/src/write_batch.dart

```diff
@@ -0,0 +1,103 @@
+// Copyright 2017, the Chromium project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of cloud_firestore;
+
+/// A [WriteBatch] is a series of write operations to be performed as one unit.
+/// 
+/// Operations done on a [WriteBatch] do not take effect until you [commit].
+/// 
+/// Once committed, no further operations can be performed on the [WriteBatch],
+/// nor can it be committed again.
+class WriteBatch {
+  WriteBatch():
+    _handle = Firestore.channel.invokeMethod(
+      'Batch#create',
+    );
+
+  Future<int> _handle;
+  final List<Future<Null>> _actions = <Future<Null>>[];
+  
+  /// Indicator to whether or not this [WriteBatch] has been committed.
+  bool committed = false;
```

Comment:
I think this should be a private property to follow the convention of the Firebase SDK